src/biogeophys/SoilStateInitTimeConstMod.F90

```diff
@@ -355,21 +355,14 @@ subroutine SoilStateInitTimeConst(bounds, soilstate_inst, nlfilename)
              soilstate_inst%tkmg_col(c,lev)   = spval
              soilstate_inst%tksatu_col(c,lev) = spval
              soilstate_inst%tkdry_col(c,lev)  = spval
-             soilstate_inst%csol_col(c,lev)= spval
+             soilstate_inst%csol_col(c,lev)   = spval
           end do
 
        else if (lun%urbpoi(l) .and. (col%itype(c) /= icol_road_perv) .and. (col%itype(c) /= icol_road_imperv) )then
 
           ! Urban Roof, sunwall, shadewall properties set to special value
-          do lev = 1,nlevgrnd
+          do lev = 1,nlevmaxurbgrnd
              soilstate_inst%watsat_col(c,lev) = spval
-             soilstate_inst%watfc_col(c,lev)  = spval
-             soilstate_inst%bsw_col(c,lev)    = spval
-             soilstate_inst%hksat_col(c,lev)  = spval
-             soilstate_inst%sucsat_col(c,lev) = spval
-             soilstate_inst%watdry_col(c,lev) = spval 
-             soilstate_inst%watopt_col(c,lev) = spval 
-             soilstate_inst%bd_col(c,lev) = spval 
              if (lev <= nlevsoi) then
                 soilstate_inst%cellsand_col(c,lev) = spval
                 soilstate_inst%cellclay_col(c,lev) = spval
```

Comment:
@olyson - On master, this block of code has been merged with the above block for wetlands and glaciers. So it's a little hard to preserve your changes as they are here. However, what I'm thinking is:
- You moved the initialization of `cellsand`, `cellclay` and `cellorg` into a loop over `nlevmaxurbgrnd` rather than `nlevgrnd`; however, since they are only set in a conditional on `lev <= nlevsoi`, this seems unnecessary (since I think we are still guaranteed to have `nlevgrnd >= nlevsoi`). So I plan to revert this change.
- You also moved the initialization of `watsat_col` into a loop over `nlevmaxurbgrnd`. That *does* seem important. But I'm thinking it's okay to initialize that to `spval` down to `nlevmaxurbgrnd` for glaciers and lakes, too - i.e., I'm thinking it shouldn't hurt to set this to `spval` down to lower than it really needs to be for those landunits.

This results in just the following diffs in this file:

```diff
diff --git a/src/biogeophys/SoilStateInitTimeConstMod.F90 b/src/biogeophys/SoilStateInitTimeConstMod.F90
index 414a8b6f1..e035a5540 100644
--- a/src/biogeophys/SoilStateInitTimeConstMod.F90
+++ b/src/biogeophys/SoilStateInitTimeConstMod.F90
@@ -158,7 +158,7 @@ subroutine SoilStateInitTimeConst(bounds, soilstate_inst, nlfilename)
     use ncdio_pio           , only : file_desc_t, ncd_io, ncd_double, ncd_int, ncd_inqvdlen
     use ncdio_pio           , only : ncd_pio_openfile, ncd_pio_closefile, ncd_inqdlen
     use clm_varpar          , only : numrad
-    use clm_varpar          , only : nlevsoi, nlevgrnd, nlevlak, nlevsoifl, nlayer, nlayert, nlevurb, nlevsno
+    use clm_varpar          , only : nlevsoi, nlevgrnd, nlevlak, nlevsoifl, nlayer, nlayert, nlevmaxurbgrnd, nlevsno
     use clm_varcon          , only : zsoi, dzsoi, zisoi, spval
     use clm_varcon          , only : secspday, pc, mu, denh2o, denice, grlnd
     use clm_varctl          , only : use_cn, use_lch4, use_fates
@@ -396,9 +396,12 @@ subroutine SoilStateInitTimeConst(bounds, soilstate_inst, nlfilename)
            (lun%urbpoi(l) .and. col%itype(c) /= icol_road_perv .and. &
                                 col%itype(c) /= icol_road_imperv)) then
 
+          do lev = 1,nlevmaxurbgrnd
+             soilstate_inst%watsat_col(c,lev) = spval
+          end do
+
           do lev = 1,nlevgrnd
              soilstate_inst%bsw_col(c,lev)    = spval
-             soilstate_inst%watsat_col(c,lev) = spval
              soilstate_inst%watfc_col(c,lev)  = spval
              soilstate_inst%hksat_col(c,lev)  = spval
              soilstate_inst%sucsat_col(c,lev) = spval
```

- [ ] Can you please confirm that this makes sense to you?

Reply:
Those suggestions make sense.